angr/analyses/forward_analysis.py

```diff
@@ -527,6 +540,14 @@ def _run_on_node(self, node, state):
         """
         The analysis routine that runs on each node in the graph.
 
+        For compatibility reasons, the variable `changed` in the returning tuple can have three values:
+        - True, means a change has occurred, the output state is not the same as the input state, and a fixed point is
+          not reached. Usually used if the analysis performs weak updates.
+        - False, means no change has occurred, and a fixed point has reached. Usually used if the analysis performs weak
```

Comment:
Doesn't mean a fixed point has been reached, right? The other nodes in the worklist could still cause a change